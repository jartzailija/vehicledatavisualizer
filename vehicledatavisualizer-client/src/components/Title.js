import React from 'react';
import { Row, Col } from 'reactstrap';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Title = ({municipality, count}) => {

  const StyledHeaderCol = styled(Col)`
    margin-top: 20px;
  `;

  const municipalityString = municipality !== '' ? ` in ${municipality}` : '';
  const titleString = <StyledHeaderCol className="col-md-12"><h2>Vehicle distribution{municipalityString}</h2></StyledHeaderCol>
  const countString = Number(count) !== Number(0) ? <div className="col-md-12"><p>{`Vehicle count is ${count}`}</p></div> : '';
  return(<Row>{titleString}{countString}</Row>);
};

Title.propTypes = {
  municipality: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired
};

export default Title;