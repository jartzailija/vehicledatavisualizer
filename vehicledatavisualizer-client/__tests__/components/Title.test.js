import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Component to be tested
import Title from '../../src/components/Title';

describe('<Title />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const params = {
        municipality: 'TestMunicipality',
        count: 123
      }
      const wrapper = shallow(<Title {...params} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });

    test('shows data correctly', () => {
      const params = {
        municipality: 'Example',
        count: 123
      }
      const wrapper = shallow(<Title {...params} />);
      const component = wrapper.dive();
      expect(component.find('h2').children().reduce((text, n) => text + n.text(), '')).toEqual('Vehicle distribution in Example');
      expect(component.find('p').childAt(0).text()).toEqual('Vehicle count is 123');
    });
  });
});