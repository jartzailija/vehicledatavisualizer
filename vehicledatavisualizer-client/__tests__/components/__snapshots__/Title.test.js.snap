// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> render() adds data correctly 1`] = `
<div
  className="row"
>
  <Styled(Col)
    className="col-md-12"
  >
    <h2>
      Vehicle distribution
       in Example
    </h2>
  </Styled(Col)>
  <div
    className="col-md-12"
  >
    <p>
      Vehicle count is 123
    </p>
  </div>
</div>
`;

exports[`<Title /> render() renders the component 1`] = `
<div
  className="row"
>
  <Styled(Col)
    className="col-md-12"
  >
    <h2>
      Vehicle distribution
       in TestMunicipality
    </h2>
  </Styled(Col)>
  <div
    className="col-md-12"
  >
    <p>
      Vehicle count is 123
    </p>
  </div>
</div>
`;
